### PR TITLE
Fixes bug in LsiModel that occurs when id2word is a Python 3 dictionary.

### PR DESCRIPTION
**Pull request by cvangysel**
The error message "you cannot concatenate dictionary keys with a list" is thrown when passing dictionary id2word to `gensim.models.lsimodel.LsiModel` in Python 3.

### CODE REVIEW

File: gensim/models/lsimodel.py [hunk 1]
```diff
@@ -298,7 +298,8 @@ def __init__(self, corpus=None, num_topics=200, id2word=None, chunksize=20000,
             self.id2word = utils.dict_from_corpus(corpus)
             self.num_terms = len(self.id2word)
         else:
-            self.num_terms = 1 + max([-1] + self.id2word.keys())
+            self.num_terms = (
+                1 + max(self.id2word.keys()) if self.id2word else -1)
```
Comment (piskvorky) [inline gensim/models/lsimodel.py]: Looks good but keep the expression on one line please.

**Comment by cvangysel** [comment]
Done.

File: gensim/models/lsimodel.py [hunk 2]
```diff
@@ -298,8 +298,7 @@ def __init__(self, corpus=None, num_topics=200, id2word=None, chunksize=20000,
             self.id2word = utils.dict_from_corpus(corpus)
             self.num_terms = len(self.id2word)
         else:
-            self.num_terms = (
-                1 + max(self.id2word.keys()) if self.id2word else -1)
+            self.num_terms = 1 + max(self.id2word.keys()) if self.id2word else -1
```
Comment (piskvorky) [inline gensim/models/lsimodel.py]: Please put brackets around the ternary expression, its reading is a little ambiguous / confusing now.

Comment (cvangysel) [inline gensim/models/lsimodel.py]: Done.

**Comment by piskvorky** [comment]
Thanks @cvangysel !

CC @tmylk LGTM.

**Comment by tmylk** [comment]
Thanks for the PR!

**Pull request merged**